RealifeTech-CoreSDK/APIV3Utilities/Tests/AuthorisationWorkerTests.swift

```diff
@@ -66,24 +66,69 @@ final class AuthorisationWorkerTests: XCTestCase {
             }).disposed(by: disposeBag)
         MockOAuthRepository.accessTokenSource.onNext(initialToken)
     }
+
+    func test_refreshAccessToken_hasRefreshTokenAndAccessTokenIsExpired_tokenIsStored() {
+        storeSpy.accessTokenValid = false
+        storeSpy.accessToken = testTokenString
+        storeSpy.refreshToken = testTokenString
+        let testOAuthToken = OAuthToken(
+            accessToken: testTokenString,
+            refreshToken: testTokenString,
+            expiresIn: testExpirySeconds,
+            tokenType: nil,
+            scope: nil)
+        let expectation = XCTestExpectation(description: "OAuth observable did emit value")
+        emitRefreshAccessToken(initialToken: testOAuthToken) { [self] token in
+            XCTAssertEqual(storeSpy.savedCredentials?.0, testTokenString)
+            XCTAssertEqual(storeSpy.savedCredentials?.1, testExpirySeconds)
+            XCTAssertEqual(storeSpy.savedCredentials?.2, testTokenString)
+            expectation.fulfill()
+        }
+        wait(for: [expectation], timeout: 0.1)
+    }
+
+    func test_refreshAccessToken_hasNoRefreshToken_returnNilObservable() {
+        storeSpy.accessTokenValid = false
```

Comment:
I can't see what part of the `sut` is being tested here. I am normally expecting a function call or something before testing if a variable is `nil`. What functionality is this testing?

Reply:
The worker generates the `sut.refreshAccessToken` observable based on the stored credentials, and it returns nil if the credentials are invalid.